freyja/utils.py

```diff
@@ -831,6 +831,76 @@ def make_dashboard(agg_df, meta_df, thresh, title, introText,
     print("Dashboard html file saved to " + outputFn)
 
 
+def collapse_barcodes(df_barcodes, df_depth, depthcutoff, locDir):
+
+    # drop low coverage sites
+    low_cov_sites = df_depth[df_depth[3].astype(int) < depthcutoff]\
+        .index.astype(str)
+    low_cov_muts = [mut for mut in df_barcodes.columns
+                    if mut[1:-1] in low_cov_sites]
+    df_barcodes = df_barcodes.drop(low_cov_muts, axis=1)
+
+    # find lineages with identical barcodes
+    duplicates = df_barcodes.groupby(df_barcodes.columns.tolist()).apply(
+        lambda x: tuple(x.index) if len(x.index) > 1 else None
+    ).dropna()
+
+    if len(duplicates) == 0:
+        return df_barcodes
+
+    # load lineage data
+    with open(os.path.join(locDir, 'data', 'lineages.yml'), 'r') as f:
+        try:
+            lineage_yml = yaml.safe_load(f)
+        except yaml.YAMLError as exc:
+            raise ValueError('Error in lineages.yml file: ' + str(exc))
+
+    lineage_data = {lineage['name']: lineage for lineage in lineage_yml}
+
+    alias_count = {}
+    merging_recomb = False
+
+    # collapse lineages into MRCA, where possible
+    for tup in duplicates:
+        pango_aliases = [lineage_data[lin]['alias']
+                         for lin in tup]
+
+        # handle case where recombinant and non-recombinant lins are merged
+        if not all([alias.startswith('B') for alias in pango_aliases])\
+           and not all([alias.startswith('X') for alias in pango_aliases]):
+            pango_aliases = [
+                alias if not alias.startswith('X') else 'B.1.1.529'
```

Comment:
I'm a bit confused here- while most recombinants to this point are Omicron descendants, there's quite a few that predate the omicron wave